File: Php/tests/AppTests/Cashgame/Running/BarModelTests.php
<?php
namespace tests\AppTests\Cashgame\Running{

	use app\Cashgame\Running\BarModel;
	use entities\Cashgame;
	use entities\GameStatus;
	use entities\Homegame;
	use tests\UnitTestCase;
	use tests\TestHelper;

	class BarModelTests extends UnitTestCase {

		/** @var Homegame */
		private $homegame;
		private $runningGame;

		function setUp(){
			$this->homegame = new Homegame();
			$runningGame = null;
		}

		function test_GameIsRunning_NoRunningGame_IsFalse(){
			$sut = $this->getSut();

			$this->assertFalse($sut->gameIsRunning);
		}

		function test_GameIsRunning_WithRunningGame_IsTrue(){
			$this->setRunningGame();

			$sut = $this->getSut();

			$this->assertTrue($sut->gameIsRunning);
		}

		function test_GameUrl_WithRunningGame_IsSet(){
			$this->setRunningGame();

			$sut = $this->getSut();

			$this->assertIsA($sut->gameUrl, 'app\Urls\RunningCashgameUrlModel');
		}

		function setRunningGame(){
			$this->runningGame = new Cashgame();
		}

		function getSut(){
			return new BarModel($this->homegame, $this->runningGame);
		}

	}

}

File: Web.Tests/ModelTests/CashgameModels/Running/BarModelTests.cs
using Core.Classes;
using NUnit.Framework;
using Web.Models.CashgameModels.Running;
using Web.Models.UrlModels;

namespace Web.Tests.ModelTests.CashgameModels.Running{

	public class BarModelTests {

		private Homegame _homegame;
		private Cashgame _runningGame;

        [SetUp]
		public void SetUp(){
			_homegame = new Homegame();
			_runningGame = null;
		}

		[Test]
        public void GameIsRunning_NoRunningGame_IsFalse(){
			var sut = GetSut();

			Assert.IsFalse(sut.GameIsRunning);
		}

		[Test]
        public void GameIsRunning_WithRunningGame_IsTrue(){
			_runningGame = new Cashgame();

			var sut = GetSut();

			Assert.IsTrue(sut.GameIsRunning);
		}

		[Test]
        public void GameUrl_WithRunningGame_IsSet(){
			_runningGame = new Cashgame();

			var sut = GetSut();

            Assert.IsInstanceOf<RunningCashgameUrlModel>(sut.GameUrl);
		}

		private BarModel GetSut(){
			return new BarModel(_homegame, _runningGame);
		}

	}

}